packages/render/src/ui/index.tsx

```diff
@@ -130,6 +130,22 @@ function createDefaultComponents<TStylingProps extends Record<string, unknown>>(
     Image(props, stylingProps) {
       const aspectRatio = props.aspectRatio.replace(":", "/");
 
+      let sanitizedSrc =
+        props.status === "frame-loading" ? undefined : props.src;
+
+      // Don't allow data URLs that are not images -- we don't want to allow arbitrary data to be loaded
+      if (
+        sanitizedSrc?.startsWith("data:") &&
+        !sanitizedSrc?.startsWith("data:image")
+      ) {
+        sanitizedSrc = "";
```

Comment:
empty string src is invalid afaik - might be better to throw some sort of error on the frame/display an error screen

Reply:
replaced with undefined for now. we don't have error handling at this level so will have to refactor

Reply:
could maybe move this check to ImageContainer and render an error there or return an error component here

Reply:
> could maybe move this check to ImageContainer and render an error there or return an error component here

would probably be best, as silent errors is not ideal + neither is logging here